in_toto/verifylib_test.go

```diff
@@ -62,16 +69,120 @@ func TestInTotoVerifyPass(t *testing.T) {
+		packageLink.Signed.(Link).Products) {
+		t.Errorf("Summary Link products don't match. Expected '%s', "+
+			"returned '%s", packageLink.Signed.(Link).Products,
+			summaryLink.Signed.(Link).Products)
+	}
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).Command,
+		packageLink.Signed.(Link).Command) {
+		t.Errorf("Summary Link command doesn't match. Expected '%s', "+
+			"returned '%s", packageLink.Signed.(Link).Command,
+			summaryLink.Signed.(Link).Command)
+	}
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).ByProducts,
+		packageLink.Signed.(Link).ByProducts) {
+		t.Errorf("Summary Link by-products don't match. Expected '%s', "+
+			"returned '%s", packageLink.Signed.(Link).ByProducts,
+			summaryLink.Signed.(Link).ByProducts)
+	}
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).ByProducts["return-value"],
+		packageLink.Signed.(Link).ByProducts["return-value"]) {
+		t.Errorf("Summary Link return value doesn't match. Expected '%s', "+
+			"returned '%s", packageLink.Signed.(Link).ByProducts["return-value"],
+			summaryLink.Signed.(Link).ByProducts["return-value"])
+	}
 }
 
-func TestInTotoVerifyLayoutDoesNotExist(t *testing.T) {
-	err := InTotoVerify("layout/does/not/exist", map[string]Key{},
-		"link/dir/does/not/matter")
-	// Asssert error type to PathError
-	if _, ok := err.(*os.PathError); ok == false {
-		t.Fail()
+func TestVerifySublayouts(t *testing.T) {
+	sublayoutName := "sub_layout"
+	var aliceKey Key
+	if err := aliceKey.LoadPublicKey("alice.pub"); err != nil {
+		t.Errorf("Unable to load Alice's public key.")
+	}
+	sublayoutDirectory := fmt.Sprintf(SublayoutLinkDirFormat, sublayoutName,
+		aliceKey.KeyId)
+	if err := os.Mkdir(sublayoutDirectory, 0700); err != nil {
+		t.Errorf("Unable to create sublayout directory.")
+	}
+	writeCodePath := path.Join(sublayoutDirectory, "write-code.776a00e2.link")
+	if err := os.Link("write-code.776a00e2.link", writeCodePath); err != nil {
+		t.Errorf("Unable to link write-code metadata.")
+	}
+	packagePath := path.Join(sublayoutDirectory, "package.2f89b927.link")
+	if err := os.Link("package.2f89b927.link", packagePath); err != nil {
+		t.Errorf("Unable to link package metadata.")
+	}
+
+	var superLayoutMb Metablock
+	if err := superLayoutMb.Load("super.layout"); err != nil {
+		t.Errorf("Unable to load super layout")
 	}
+
+	stepsMetadata := make(map[string]map[string]Metablock)
+	var err error
+	if stepsMetadata, err = LoadLinksForLayout(superLayoutMb.Signed.(Layout),
+		"."); err != nil {
+		t.Errorf("Unable to load link metadata for super layout.")
+	}
+
+	stepsMetadataVerified := make(map[string]map[string]Metablock)
+	if stepsMetadataVerified, err = VerifyLinkSignatureThesholds(
+		superLayoutMb.Signed.(Layout), stepsMetadata); err != nil {
+		t.Errorf("Unable to verify link threshold values.")
+	}
+
+	if _, err := VerifySublayouts(superLayoutMb.Signed.(Layout),
+		stepsMetadataVerified, "."); err != nil {
+		t.Errorf("Unable to verify sublayouts.")
+	}
```

Comment:
It would also be nice to take a look at the result to see if it returns the right thing.

Reply:
What would such a check look like? Should we verify the data in the `map`?